IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m

```diff
@@ -77,7 +78,7 @@ - (NSDictionary *)jsonDictionary
     json[MSID_BROKER_WPJ_STATUS_KEY] = [self wpjStatusStringFromEnum:self.wpjStatus];
     json[MSID_BROKER_BROKER_VERSION_KEY] = self.brokerVersion;
     json[MSID_ADDITIONAL_EXTENSION_DATA_KEY] = [self.additionalExtensionData msidJSONSerializeWithContext:nil];
-    
+    json[MSID_BROKER_MDM_ID_KEY] = self.mdmId;
```

Comment:
Do we still assign if mdmid is nil?

Reply:
Do we always look for not null before assigning to json dict ? I wasnt sure

Reply:
Added the nil check before updating extraDeviceInformation in MSAL